### PR TITLE
chore: allowing support of Symfony  7

### PR DESCRIPTION
**Pull request by pauljosephkrogulec**
##  allowing support of Symfony  7

I am targeting this branch, because is for allowing the possibility of use in symfony 7 project and don't create a BC-break

## Changelog

Allowing support of Symfony 7  (@pauljosephkrogulec)

```markdown
### Added
- Support of Symfony 7
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,14 +21,14 @@
     "homepage": "https://docs.sonata-project.org/projects/form-extensions",
     "require": {
         "php": "^8.0",
-        "symfony/event-dispatcher": "^5.4 || ^6.2",
-        "symfony/form": "^5.4 || ^6.2",
-        "symfony/options-resolver": "^5.4 || ^6.2",
-        "symfony/property-access": "^5.4 || ^6.2",
-        "symfony/security-csrf": "^5.4 || ^6.2",
-        "symfony/translation": "^5.4 || ^6.2",
+        "symfony/event-dispatcher": "^5.4 || ^6.2 || 7.0.*",
```
Comment (VincentLanglet) [inline composer.json]:
```suggestion
        "symfony/event-dispatcher": "^5.4 || ^6.2 || ^7.0",
```
To allow futur Symfony minor version.

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

**Comment by VincentLanglet** [comment]
HI @pauljosephkrogulec

In order to support Symfony 7 we need to be able to run CI with it, like I tried 
here: https://github.com/sonata-project/EntityAuditBundle/pull/592/files#diff-245392b692a50c38ecab4381b118862db514035c10983f3bd4f4b7f1f4be4692R60-R64

But, currently SF 7 doesn't run on our build because of PHP-CS-Fixer, cf
https://github.com/sonata-project/EntityAuditBundle/actions/runs/6954735614/job/18922120160?pr=592

Let's wait an answer on https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7431#issuecomment-1822341952

**Review by VincentLanglet — CHANGES_REQUESTED** [review]
Can you update the github actions like in https://github.com/sonata-project/twig-extensions/pull/392 in order to be sure that SF 7 can be installed ? Thanks

**Comment by VincentLanglet** [comment]
You'll need
```
"matthiasnoback/symfony-config-test": "^4.2 || ^5.1",
"matthiasnoback/symfony-dependency-injection-test": "^4.0 || ^5.0",
```

**Comment by VincentLanglet** [comment]
There is one test failure:
```
1) Sonata\Form\Tests\Type\FormChoiceWidgetTest::testLabelRendering
Failed asserting that '<div id="choice"><input type="checkbox" id="choice_0" name="choice[]" value="0"><label for="choice_0">[trans]some[/trans]</label><input type="checkbox" id="choice_1" name="choice[]" value="1"><label for="choice_1">[trans]choices[/trans]</label></div>' contains "<div id="choice"><input type="checkbox" id="choice_0" name="choice[]" value="0" /><label for="choice_0">[trans]some[/trans]</label><input type="checkbox" id="choice_1" name="choice[]" value="1" /><label for="choice_1">[trans]choices[/trans]</label></div>".
```

We'll need to understand what's changed in SF7

**Comment by pauljosephkrogulec** [comment]
> There is one test failure:
> 
> ```
> 1) Sonata\Form\Tests\Type\FormChoiceWidgetTest::testLabelRendering
> Failed asserting that '<div id="choice"><input type="checkbox" id="choice_0" name="choice[]" value="0"><label for="choice_0">[trans]some[/trans]</label><input type="checkbox" id="choice_1" name="choice[]" value="1"><label for="choice_1">[trans]choices[/trans]</label></div>' contains "<div id="choice"><input type="checkbox" id="choice_0" name="choice[]" value="0" /><label for="choice_0">[trans]some[/trans]</label><input type="checkbox" id="choice_1" name="choice[]" value="1" /><label for="choice_1">[trans]choices[/trans]</label></div>".
> ```
> 
> We'll need to understand what's changed in SF7

The error is cause by the adding of / in the of the input (/> => >)

**Comment by VincentLanglet** [comment]
> The error is cause by the adding of / in the of the input (/> => >)

Sure, but removing the `/` won't solve our issue. It will solve the error on SF7 but tests on SF5-6 will fail.
Also, does the html is still valid without the `/` ? Maybe Symfony has a regression which need to be reported...

**Comment by pauljosephkrogulec** [comment]
> > The error is cause by the adding of / in the of the input (/> => >)
> 
> Sure, but removing the `/` won't solve our issue. It will solve the error on SF7 but tests on SF5-6 will fail. Also, does the html is still valid without the `/` ? Maybe Symfony has a regression which need to be reported...

The problem come from the bundle symfony/twig-bridge https://github.com/symfony/twig-bridge/commit/dcb5bb80aade1571fb28a4a57a0a65100c1252bd
I have a solution with a if but i don't know if it's a good one using `Kernel::VERSION`

**Comment by VincentLanglet** [comment]
> The problem come from the bundle symfony/twig-bridge [symfony/twig-bridge@dcb5bb8](https://github.com/symfony/twig-bridge/commit/dcb5bb80aade1571fb28a4a57a0a65100c1252bd) 

thanks for the research. So it's a "feature" from Symfony.

> I have a solution with a if but i don't know if good one using `Kernel::VERSION`

I would be ok with this.

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**